### DISCUSSION TITLE
Improving R-Instat for the analysis of agricultural research data

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#6620] [Ideas]
This has always been one of the application areas we would like R-Instat to address.  We also plan a workshop on analysis  for CCRP in perhaps January 2022.  This would be initially for participants in West Africa and would be in French.  So it will benefit from R-Instat now being (sort of) in French and that aspect will be much appreciated by the participants.

In the olden days we promoted Genstat and that is still the obvious package.  They used to have a free version - called Genstat Discovery,  And the fact this was withdrawn was part of the impetus to develop R-Instat.  In the original 2015 video @volloholic discussing the fact that researchers in Burkina Faso didn't have an obvious package was part of the start of R-Instat for the group we are considering here.)

These developments are slightly different to our current push for R-Instat in relation to introducing data science.  The overlap will be through some interesting datasets.  We should look for examples of complementary survey datasets as secondary data to hlop the links with data science.  

One more bit of background - who is the target audience and how do the proposed additions fit into the current version of R-Instat.  The target audience includes staff from the International Agricultural centres.  There are many of these.  I used to work for ICRISAT (research in semi-arid tropics) and still have contacts and we will have some data from there.  Ric Coe still works for ICRAF (Agroforestry).  The standard package in R is called agricolae and most of the data sets are from CIP (potato research).  CIMMYT (maize) is another large centre and IRRI (rice) is one of the oldest.  It has a West African partner called AfricaRice (used to be WARDS, etc.  These are all part of a group of 15 centres called CGIAR.  

There are more international centres.  For example [Crops for the Future](https://en.wikipedia.org/wiki/Crops_for_the_Future_(CFF)) and ICIPE (insects).  It is part of another grouping of international centres called [AIRCA](https://en.wikipedia.org/wiki/Association_of_International_Research_and_Development_Centers_for_Agriculture).   

One reason for associating with these centres is that they are generally very positive about sharing data freely.  They all work closely with NARS - a second audience for these resources.  So each country also has its own national Agricultural Research organisation.  For example, this is KARI in Kenya, NARO in Uganda, IER in Mali, INERA in Burka Faso, and INRAN in Niger.  

Our work in West Africa is also with Farmer's organisations and NGOs and they are also increasingly becoming involved in Research.  Then, of course, many Universities have a faculty of Agriculture and their staff and students could benefit from what we produce!

So, what's our starting point:
a) There is a "standard package" and this is called `agricolae`.  This is already in R-Instat as are all the packages it depends on.  It includes a lot of useful datasets, and these are mainly from the international potato centre, CIP.  Some are just to illustrate analyses, while others are larger and more general.  The approach in this package is pretty "traditional" with specialised analyses for the common designs and a lot of concentration on multiple comparisons - which remains a contentious issue. Here is a useful paper about agricolae: 
[peerj-preprints-1404.pdf](https://github.com/africanmathsinitiative/R-Instat/files/6835976/peerj-preprints-1404.pdf)

b) There is a package called agriTutorial described here: 
[agriTutorialVignette.pdf](https://github.com/africanmathsinitiative/R-Instat/files/6835977/agriTutorialVignette.pdf)

This describes the analysis of 5 datasets.  It uses a package called lmerTest, which uses lme4 (already installed).  This looks interesting to install and go through, though the 5 datasets are all what I call "TextBook" example. I find that textbook examples usually only measure one thing - often yield.  And that's the situation with all these examples.  In contrast, almost all real experiments measure many things - though they often don't analyse them all!

So what should we do?  We do want to start from where our main audience is now.  So we need to include good analyses of single variables in conventional ways.  In addition we also believe in "Options by Context", namely that only few problems result in a "silver bullet", e.g. a best variety or a best treatment.  Usually what is sensible to adopt depends on the context (soil, rain, fertility, work, etc).  So we need to ask for "best for which contexts", or even more generally "what are suggested options for each context.

@volloholic has suggested we could usefully have a special menu.  Also his Options by Context menu is (unnecessarily) provocative and could be renamed.  What's the new name?  Trials or Experiments or Agriculture?  It is for experimental data.  An experiment has 3 components, namly Layout, Treatments and Measurements.  Here roughly Layout is largely the context, while treatments gives the options we are investigating.  The measurements are partly more context and partly "results".  

Data sets to start with will include the 5 from the tutorial and some of those in agricolae.  We will add some from ICRISAT soon.

**Comment by rdstern** [comment]
We also need to add the Fuma Gaskiya set  as am impressive example that also shows options by context clearly.  And perhaps we could also include an example from Tomato and the millet game.  Not silly to include some data from a simulation.  

Perhaps we might also include an example of data from a crop simulation model?

There are topics to discuss here!

**Reply by fran2or** [reply]
Quite detailed and relevant background information. @IssoufKon and I (with @volloholic ) have also had some brief discussions in the conceptual development of this menu bearing in mind the target audience. Unfortunately, the detailed background information you shared had not been fully considered in our initial ideas for conceptualizing this menu given we both missed the discussion thread. After  going through the discussion thread and resources shared, we think there are relevant ideas that we could take on to improve our design conceptualization. We look forward to concretizing our ideas  through discussions with everyone  during the sprint week and hopefully have a conceptual  menu design ready at the end of the sprint.

**Comment by fran2or** [comment]
After several discussions with @IssoufKon and @volloholic, the attached document is our current conceptual proposal for the entire design of the Agricultural Research menu, now the Experimental menu. Designing of new dialogs is still on-going for reviews. This document will be reviewed as and when is needed.
[Experimental Menu Design.docx](https://github.com/africanmathsinitiative/R-Instat/files/7261257/Experimental.Menu.Design.docx)
